oepcie/yuanphil_impl/devices.h

```diff
@@ -0,0 +1,43 @@
+#ifndef _INCL_GUARD_THREE
+#define _INCL_GUARD_THREE
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <sys/types.h> 
+#include <sys/stat.h> 
+#include <fcntl.h> 
+#include <unistd.h>
+#include <errno.h>
+#include <string.h>
+
+#define MAX_DEVICE_ID 3
+#define DEVICE_TYPES 4
+
+typedef struct device{
+	size_t read_offset;
+	size_t read_size;
+	size_t write_offset;
+	size_t write_size;
+
+	int id;
+} device_t;
+
+/*
+ * This is the way we settled on to hardcode device information. You are 
+ * initializing a global array of devices with hardcoded information for every
+ * device. You can get a devices basic info by indexing into the DEVICES array
+ * with the devices id. In practice, this will be read by the header file.
+ */
+
+static device_t DEVICES[DEVICE_TYPES] = {
+	{.read_offset = 0, .read_size = (16 * 32), 
+		.write_offset = 0, .write_size = 0, .id = 0}, // RHD2032
+	{.read_offset = 0, .read_size = (16 * 64), 
```

Comment:
Would you provide us with a list of devices that we should include in the program? Ideally with their data sizes as well, but if that is not readily available we can look up that information on the data sheets.